framework/src/source/TestRunner.bs

```diff
@@ -49,6 +45,10 @@ namespace rooibos
     '  */
     public function run()
 
+      for each reporter in m.testReporters
+        reporter.onBegin()
```

Comment:
None of the reporter hooks are required, so we need to guard this with an "is onBegin defined" check before calling it.

Reply:
Also, please pass `{}` for now. Every reporter hook will take an `event` object. I have no idea what stuff we're going to put in there for now, so an empty AA for now is acceptable.

Reply:
All valid points. I initially thought that a custom reporter would always extend from `BaseTestReporter`, but after reading your comment I realise it might not be the case, and that not all users might be fully using brighterscript in their projects.

With that said, these changes are now done!